pkg/controllers/jobset_controller_test.go

```diff
@@ -700,6 +704,223 @@ func TestUpdateConditions(t *testing.T) {
 	}
 }
 
+func TestCalculateReplicatedJobStatuses(t *testing.T) {
+	var (
+		jobSetName = "test-jobset"
+		ns         = "default"
+	)
+	tests := []struct {
+		name     string
+		js       *jobset.JobSet
+		jobs     childJobs
+		expected []jobset.ReplicatedJobStatus
+	}{
+		{
+			name: "all jobs are ready, no succeeded jobs",
+			js: testutils.MakeJobSet(jobSetName, ns).
+				ReplicatedJob(testutils.MakeReplicatedJob("replicated-job-1").
+					Job(testutils.MakeJobTemplate("test-job", ns).Obj()).
+					Replicas(1).
+					Obj()).
+				ReplicatedJob(testutils.MakeReplicatedJob("replicated-job-2").
+					Job(testutils.MakeJobTemplate("test-job", ns).Obj()).
+					Replicas(3).
+					Obj()).Obj(),
+			jobs: childJobs{
+				active: []*batchv1.Job{
+					makeJob(&makeJobArgs{
+						jobSetName:        jobSetName,
+						replicatedJobName: "replicated-job-1",
+						jobName:           "test-jobset-replicated-job-1-test-job-0",
+						ns:                ns,
+						replicas:          1,
+						jobIdx:            0,
+						parallelism:       pointer.Int32(1),
+						completions:       pointer.Int32(1),
+						ready:             pointer.Int32(1),
+						succeeded:         1}).Suspend(false).Obj(),
+					makeJob(&makeJobArgs{
+						jobSetName:        jobSetName,
+						replicatedJobName: "replicated-job-2",
+						jobName:           "test-jobset-replicated-job-2-test-job-0",
+						ns:                ns,
+						replicas:          3,
+						jobIdx:            0,
+						parallelism:       pointer.Int32(5),
+						ready:             pointer.Int32(2),
+						succeeded:         3}).Obj(),
+					makeJob(&makeJobArgs{
+						jobSetName:        jobSetName,
+						replicatedJobName: "replicated-job-2",
+						jobName:           "test-jobset-replicated-job-2-test-job-1",
+						ns:                ns,
+						replicas:          3,
+						jobIdx:            0,
+						parallelism:       pointer.Int32(3),
+						completions:       pointer.Int32(2),
+						ready:             pointer.Int32(1),
+						succeeded:         1}).Obj(),
+					makeJob(&makeJobArgs{
+						jobSetName:        jobSetName,
+						replicatedJobName: "replicated-job-2",
+						jobName:           "test-jobset-replicated-job-2-test-job-2",
+						ns:                ns,
+						replicas:          3,
+						jobIdx:            0,
+						parallelism:       pointer.Int32(2),
+						completions:       pointer.Int32(3),
+						ready:             pointer.Int32(2),
+						succeeded:         1}).Obj(),
```

Comment:
can you add another job that is not ready (ready+succeeded less than completions/parallelism)